DEPENDENCIES.md

```diff
@@ -0,0 +1,71 @@
+# Upgrading Go Version and Dependencies in Kubeflow
+
+This guide outlines the steps to upgrade the Go version and dependencies in the Kubeflow project, including the necessary changes in the red-hat-data-services/kubeflow fork.
+
+## Upgrading Go Version
+
+Upgrading the Go version should be done in a separate PR to isolate the changes and make review easier.
+
+> [!IMPORTANT]  
+> Images are to be built in the [ubi8/go-toolset](https://catalog.redhat.com/software/containers/ubi8/go-toolset/5ce8713aac3db925c03774d1) container.
+> It contains a customized FIPS-compatible version of Go, that however lags behind the latest upstream Go version.
+> Always use a Go version that has a supporting go-toolset image available.
+
+1. Begin by reading [Go release notes](https://go.dev/doc/devel/release) to identify potential incompatibilities.
+
+2. Update the Go version in the following files:
+   - `components/**` `/` `go.mod`: Update the `go` directive at the top of the file.
+   - `components/**` `/` `Dockerfile`: Update the Go version in the base image.
+
+3. Run the following commands to update and verify the build:
+
+    ```shell
+    go mod tidy
+    make test
+    ```
+
+4. Commit these changes and create a pull request for the Go version upgrade.
+
+   > [!WARNING]  
+   > Use the `Manifest List Digest` and not the `Image Digest` when locating sha256 in the Red Hat Image Catalog entry.
+
+5. After merging the Go upgrade PR, update the fork at https://github.com/red-hat-data-services/kubeflow:
+   - Locate all `Dockerfile.konflux` files in the repository.
+   - Update the Go version in each of these files to match the new version.
+   - Refer to the [ubi8/go-toolset](https://catalog.redhat.com/software/containers/ubi8/go-toolset/5ce8713aac3db925c03774d1) in Red Hat image catalog to locate `sha256` hash
+   - Create a pull request in the fork repository with these changes.
+
+6. Review CI/CD configuration files (especially openshift/release OCP-CI yamls) that specify a Go version.
```

Comment:
> Hmm, if the openshift-ci is able to grab the build_root from the repo, I would vote for the second option so that there is less places to update this.

Do we need to remove point 6. or we still keep it?

Reply:
let's keep it here until we update the openshift-ci (if ever)